### PR TITLE
Updates to local document operations in migration guide

### PR DESCRIPTION
**Pull request by emlaver**
## Checklist

- [x] Tick to sign-off your agreement to the [Developer Certificate of Origin (DCO) 1.1](../blob/master/DCO1.1.txt)
- [x] Added tests for code changes _or_ test/build only changes
- [x] Updated the change log file (`CHANGES.md`) _or_ test/build only changes
- [x] Completed the PR template below:

## Description
Migration guide updates:
- Replaced `create` with `save` local doc operation. The `save` operation with use `create` under-the-covers if the document doesn't exist. 
- Group together local doc operations
- Fix API doc link for local doc delete operation

### CODE REVIEW

File: MIGRATION.md [hunk 1]
```diff
@@ -82,7 +83,6 @@ The table below contains a list of `python-cloudant` functions and the `cloudant
 |`db.get_security_document()/db.security_document()`|[getSecurity](https://cloud.ibm.com/apidocs/cloudant?code=python#getsecurity)|
 |`db.share_database()`|[putSecurity](https://cloud.ibm.com/apidocs/cloudant?code=python#putsecurity)|
 |`db.shards()`|[getShardsInformation](https://cloud.ibm.com/apidocs/cloudant?code=python#getshardsinformation)|
-|`Document(db, 'docid').delete()`|[getLocalDocument](https://cloud.ibm.com/apidocs/cloudant?code=python#getlocaldocument)|
```
Comment (vmatyusGitHub) [inline MIGRATION.md:85]: This should be a `deleteDocument`.

**Review by bessbd — APPROVED** [review]

File: MIGRATION.md (hunk 1 above)
Comment (emlaver) [reply MIGRATION.md:85]: @vmatyusGitHub I moved this above so the local operations are grouped together: https://github.com/cloudant/python-cloudant/pull/489/files#diff-c1603ee6a674bb75255aa448057630506153e363297fd58668e6679f10909bc6R75

Comment (vmatyusGitHub) [reply MIGRATION.md:85]: Yes, I see that, but please note that a simple document delete is missing from this table: https://cloud.ibm.com/apidocs/cloudant?code=python#deletedocument

Comment (emlaver) [reply MIGRATION.md:85]: ah yes, I'll add that!

Comment (emlaver) [reply MIGRATION.md:85]: Added in https://github.com/cloudant/python-cloudant/pull/489/commits/2baf07d4878c09b2e329e101056888ae22c54e1b

**Review by vmatyusGitHub — APPROVED** [review]
No more comment from me, +1

**Pull request merged**